flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2338,69 +2340,76 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     //Needs PR Merge #3082
-    trySetThrustReductionAccelerationAltitude(s) {
+    async trySetThrustReductionAccelerationAltitude(s) {
         if (s === FMCMainDisplay.clrValue) {
             this.thrustReductionAltitudeIsPilotEntered = false;
             this.accelerationAltitudeIsPilotEntered = false;
-            CDUPerformancePage.UpdateThrRedAccFromOrigin(this, true, true);
+            CDUPerformancePage.UpdateThrRedAccFromOrigin(this);
             return true;
         }
 
         const origin = this.flightPlanManager.getOrigin();
-        const elevation = origin ? origin.altitudeinFP : 0;
+
+        let elevation = SimVar.GetSimVarValue("GROUND ALTITUDE", "feet");
+        if (origin) {
+            elevation = await this.facilityLoader.GetAirportFieldElevation(origin.icao);
+        }
+
         const minimumAltitude = elevation + 400;
 
-        let newThrRedAlt = null;
-        let newAccAlt = null;
+        let checkThrRedAlt = this.thrustReductionAltitude;
+        let checkAccAlt = this.accelerationAltitude;
 
         let [thrRedAlt, accAlt] = s.split("/");
 
-        if (thrRedAlt && thrRedAlt.length > 0) {
-            if (!/^\d{3,5}$/.test(thrRedAlt)) {
+        if (thrRedAlt && thrRedAlt.length > 0 || accAlt && accAlt.length > 0) {
+
+            if (thrRedAlt && thrRedAlt.length > 0 && !/^\d{3,5}$/.test(thrRedAlt)) {
                 this.addNewMessage(NXSystemMessages.formatError);
                 return false;
+            } else if (thrRedAlt && thrRedAlt.length > 0) {
+                thrRedAlt = parseInt(thrRedAlt);
+                thrRedAlt = Math.round(thrRedAlt / 10) * 10;
+                checkThrRedAlt = thrRedAlt;
             }
 
-            thrRedAlt = parseInt(thrRedAlt);
-            thrRedAlt = Math.round(thrRedAlt / 10) * 10;
-            if (thrRedAlt < minimumAltitude || thrRedAlt > 45000) {
-                this.addNewMessage(NXSystemMessages.entryOutOfRange);
+            if (accAlt && accAlt.length > 0 && !/^\d{3,5}$/.test(accAlt)) {
+                this.addNewMessage(NXSystemMessages.formatError);
                 return false;
+            } else if (accAlt && accAlt.length > 0) {
+                accAlt = parseInt(accAlt);
+                accAlt = Math.round(accAlt / 10) * 10;
+                checkAccAlt = accAlt;
             }
 
-            newThrRedAlt = thrRedAlt;
         }
```

Comment:
I think this block is a bit convoluted. I believe we can clean this up a bit by removing some redundant null checks on `thrRedAlt` and `accAlt` for example. Furthermore, it seems that this block does not check whether the altitudes are in the required range, and that this check is done further down. Are you sure this is intended?